crates/sui-framework/packages/move-stdlib/sources/vector.move

```diff
@@ -257,4 +272,77 @@ module std::vector {
             true
         }
     }
+
+    /// Zip two vectors `v1` and `v2` by applying the function `f` to each pair of elements.
+    /// Consumes both vectors.
```

Comment:
I wouldn't maybe use the verb "zip" in this way, since to me `zip` would make `vector<(T1, T2)>`

So maybe something like 
```suggestion
    /// Destroys two vectors `v1` and `v2` by calling `f` to each pair of elements.
    /// Consumes both vectors.
```